sqlobject/src/test/java/org/jdbi/v3/sqlobject/TestReturningMap.java

```diff
@@ -0,0 +1,252 @@
+        dao.insert("foo", "123");
+        dao.insert("bar", "xyz");
+
+        assertThat(dao.getAll()).containsOnly(
+                entry("foo", "123"),
+                entry("bar", "xyz"));
+    }
+
+    public interface KeyValueDao {
+        @SqlUpdate("create table config (key varchar, value varchar)")
+        void createTable();
+
+        @SqlUpdate("insert into config (key, value) values (:key, :value)")
+        void insert(String key, String value);
+
+        @SqlQuery("select key, value from config")
+        @KeyColumn("key")
+        @ValueColumn("value")
+        Map<String, String> getAll();
+    }
+
+    @Test
+    public void uniqueIndex() {
+        UniqueIndexDao dao = h.attach(UniqueIndexDao.class);
+
+        dao.createTable();
+
+        dao.insert(
+                new User(1, "alice"),
+                new User(2, "bob"),
+                new User(3, "cathy"),
+                new User(4, "dilbert"));
+
+        assertThat(dao.getAll()).containsOnly(
+                entry(1, new User(1, "alice")),
+                entry(2, new User(2, "bob")),
+                entry(3, new User(3, "cathy")),
+                entry(4, new User(4, "dilbert")));
+    }
+
+    public interface UniqueIndexDao {
+        @SqlUpdate("create table user (id int, name varchar)")
+        void createTable();
+
+        @SqlBatch("insert into user (id, name) values (:id, :name)")
+        void insert(@BindBean User... users);
+
+        @SqlQuery("select * from user")
+        @KeyColumn("id")
+        @RegisterConstructorMapper(User.class)
+        Map<Integer, User> getAll();
+    }
+
+    @Test
+    public void joinRow() {
+        JoinRowDao dao = h.attach(JoinRowDao.class);
+        dao.createUserTable();
+        dao.createPhoneTable();
+
+        dao.insertUsers(
+                new User(1, "alice"),
+                new User(2, "bob"),
+                new User(3, "cathy"));
+        dao.insertPhone(1, new Phone(10, "555-0001"));
+        dao.insertPhone(2, new Phone(20, "555-0002"));
+        dao.insertPhone(3, new Phone(30, "555-0003"));
+
+        assertThat(dao.getMap()).containsOnly(
+                entry(new User(1, "alice"), new Phone(10, "555-0001")),
+                entry(new User(2, "bob"), new Phone(20, "555-0002")),
+                entry(new User(3, "cathy"), new Phone(30, "555-0003")));
+    }
+
+    @Test
+    public void multimapJoin() {
+        JoinRowDao dao = h.attach(JoinRowDao.class);
+        dao.createUserTable();
+        dao.createPhoneTable();
+
+        dao.insertUsers(
+                new User(1, "alice"),
+                new User(2, "bob"),
+                new User(3, "cathy"));
+        dao.insertPhone(1, new Phone(10, "555-0001"), new Phone(11, "555-0021"));
+        dao.insertPhone(2, new Phone(20, "555-0002"), new Phone(21, "555-0022"));
+        dao.insertPhone(3, new Phone(30, "555-0003"), new Phone(31, "555-0023"));
+
+        assertThat(dao.getMultimap()).hasSameEntriesAs(
```

Comment:
Can't do this one. Multimap.equals() contract is basically: this.asMap().equals(that.asMap(). So this common fixture wouldn't work across ListMultimap and SetMultimap.